File: Data Structures & Algorithms/anagram-groups/submission-0.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> anagram = new HashMap<>();
        for(String i : strs) {
            char[] charArray = i.toCharArray();
            Arrays.sort(charArray);
            String sortedstr = new String(charArray);
            anagram.putIfAbsent(sortedstr, new ArrayList<>());
            anagram.get(sortedstr).add(i);
        }
        return new ArrayList(anagram.values());
    }
}

File: Data Structures & Algorithms/string-encode-and-decode/submission-0.py
class Solution:

    def encode(self, strs: List[str]) -> str:
        msg = ""
        for s in strs:
            msg += str(len(s)) + "#" + s
            print(msg)
        return msg

    def decode(self, s: str) -> List[str]:
        msg = []
        i = 0
        while i<len(s):
            j = i
            while s[j] != '#':
                j+= 1
            length = int(s[i:j])
            i = j + 1
            j = i + length
            msg.append(s[i:j])
            i = j
        return msg